docs/core-team-process.md

```diff
@@ -0,0 +1,77 @@
+* Respect for **differing individual preferences** in social needs and tolerance for practices like pair programming.
+* Acknowledgement that **we are distributed geographically** and the differences in timezone and daily schedules that that implies.
+* **Continuous improvement** to find what works best for the team we are today and for the immediate problem at hand, and to adjust as both of these change fluidly.
+
+## Organization
+
+When we plan, we choose to pursue _a single task_ as a single team, rather than distributing tasks among ourselves from a queue and working on independent tasks in parallel. This is intended to increase the amount and quality of communication we can share in chat and in synchronous meetings: it's much easier to maintain an ongoing technical conversation when all participants share the mental context of a unified goal.
+
+This does not mean that we all pair program all the time. We do get value from pair programming but this is not always practical or desirable. Pair programming may be chosen independently from the methods below -- functionally, the pair becomes one "developer" in any of the descriptions.
+
+### 1. Seams
+
+Divide the issue at hand among the team along the abstraction layers in our codebase. Each developer continuously negotiates the interface with neighboring layers by an active Slack conversation, correcting their direction based on feedback. Developers push their work as commits to a single shared branch, documenting and coordinating overall progress in a shared pull request.
+
+> Example: developer A implements changes to the model, developer B implements the view component, and developer C implements the controller methods. Developer B writes code as though the model and controller are already complete, using sinon mocks for tests and communicating the view's needs as they arise. Developer C proceeds similarly with the controller methods. Developer A gives feedback on the feasibility of requested model functionality from both A and B and negotiates method names and property names. When developer C leaves for the day or takes time off, developers A and B proceed, leaving asynchronous notes for developer C as pull request comments for them to catch up on when they come back online.
+
+:+1: _Advantages:_
+
+* Encourages high-touch, continuous conversation involving and relevant to the full team.
+* Resilient to time off and asynchronicity.
+* Minimizes the need to context switch up and down abstraction layers while working.
+
+:-1: _Disadvantages:_
+
+* Diminishes variety of work done by any individual developer, which could become boring.
+* Reduces the familiarity developed by any single developer to a single abstraction layer within the codebase.
+* Timing may become difficult. It's possible that one "seam" may take much more time to implement than the others, which could lead to a bottleneck.
+* Some efforts will not be decomposable into easily identified seams for division of labor.
+
```

Comment:
🤔 This approach relies heavily on this to be true:
>Each developer continuously negotiates the interface with neighboring layers by an active Slack conversation, correcting their direction based on feedback.

-- with `continuous negotiation` being the key point, which makes me think this is not very timezone-friendly. 

I am imagining a scenario where developers A, B, and C agree on a certain direction to finish a task. Developer A (6 hours ahead of the others) goes ahead with the agreed upon direction for the model layer. Developer B starts their day 6 hours later, upon making changes to the view component, realizes the original appraoch wouldn't work because of Reasons™️; they let developer A knows, and the team renegotiates on the direction, but that may result in having to throw out the work developer A has already done.

Reply:
Yeah, you're right. When I'm working alone I tend to iterate pretty quickly up and down abstraction layers. As soon as we're stretched across timezones we risk lots of running down blind alleys.

Here's what's in my head right now: each set of developers who are online synchronously can divide work as we divide in Seams; as that set changes when people come online and drop offline, we do Handoffs to pass context along.

For example, say:

1. You come online first in your morning and work solo for a few hours, shifting up and down the stack just like you would otherwise.
2. When I come online, I get caught up on the work you've pushed so far and chat with you to sync up on progress. We divvy up areas of work to focus on for the next few hours, chatting in Slack as we go.
3. @annthurium and @kuychaco come online next. We bring them up to speed and subdivide further. Maybe they pair on the view work while we work on the model and controller.
4. When you're done for the day, you summarize how far you got on your bit. One of the three of us catches up, picks up where you left off, and keeps it going. I do the same when I log off.
5. When @annthurium and @kuychaco are finishing up they leave a quick writeup of their collective progress.
7. The next morning, you read the diff and the writeup and get traction on continuing through your day.
8. ...and repeat. ♻️ 

Does that sound rational?

Reply:
it does!